mate-panel/panel-toplevel.c

```diff
@@ -1269,11 +1245,77 @@ static void panel_toplevel_update_buttons_showing(PanelToplevel* toplevel)
 	}
 }
 
+/* force set hide button size on panel size < 30px */
+static void panel_toplevel_update_hide_buttons_size(GtkWidget* button, int panel_size)
+{
+
+	GtkStyleContext *context = NULL;
+	context = gtk_widget_get_style_context(button);
+	gtk_style_context_add_class(context, "panel-button");
+
+	/* memory is managed by gtk */
+	GtkCssProvider *css_provider = NULL;
+	css_provider = gtk_css_provider_get_default ();
+
+	/* get arrow image */
+	GtkWidget *arrow = NULL;
+	arrow = gtk_bin_get_child (GTK_BIN (button));
+
+	/* get defaults from theme */
+	GtkSettings * settings = NULL;
+	settings = gtk_settings_get_default ();
+	gchar *gtk_theme_name = NULL;
+	g_object_get (settings, "gtk-theme-name", &gtk_theme_name, NULL);
+	css_provider = gtk_css_provider_get_named (gtk_theme_name, NULL);
+	g_free (gtk_theme_name);
+
+
+	/* set custom css by adding our custom code to the default theme css
+	 *
+	 * NOTE that contriary to the documentation:
+	 * https://developer.gnome.org/gtk3/stable/GtkCssProvider.html#gtk-css-provider-load-from-data
+	 * the previously loaded theme is NOT cleared from the css_provider. (reason unknown)
```

Comment:
What? Did quick test and it works... Also did you know that mentioned function returns boolean? You could use that to check if it executed successfully... Also that function has GError that you could use to get error message in case it fails.